osu.Game.Tests/Visual/Gameplay/TestSceneOverlayActivation.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Linq;
+using NUnit.Framework;
+using osu.Game.Configuration;
+using osu.Game.Overlays;
+using osu.Game.Rulesets;
+
+namespace osu.Game.Tests.Visual.Gameplay
+{
+    public class TestSceneOverlayActivation : OsuPlayerTestScene
+    {
+        private OverlayTestPlayer testPlayer;
+
+        public override void SetUpSteps()
+        {
+            AddStep("disable overlay activation during gameplay", () => LocalConfig.Set(OsuSetting.GameplayDisableOverlayActivation, true));
+            base.SetUpSteps();
+        }
+
+        [Test]
+        public void TestGameplayOverlayActivation()
+        {
+            AddAssert("activation mode is disabled", () => testPlayer.OverlayActivationMode == OverlayActivation.Disabled);
```

Comment:
why are the tests only covering the disabled case?

Reply:
The next test under this one covers disabling the setting.

Reply:
ah, i see. got confused by the `SetUpStep` setting one state. maybe not the best way to structure tests (move the set into each test instead, or via `[TestCase(bool)]`)

Reply:
I pushed a commit to make the test steps hopefully clearer